torch/csrc/autograd/record_function_ops.cpp

```diff
@@ -68,7 +66,11 @@ void _call_end_callbacks_on_fut(
         at::ThreadLocalStateGuard g(tls_state);
         auto& rec = getRecordFunctionFromTensor(handle);
         rec._end();
-      });
+        return fut->constValue();
```

Comment:
Let's add a comment to explain why we need to forward the return value here. IIUC, it is because `_invoke_rpc` actually returns this future?

Reply:
Yeah, we get the future which has the actual result from `_invoke_rpc`. Then, if profiling we run this, and we want to ensure that the correct value of the future from `_invoke_rpc` is propagated. Will add this to the comment